### PR TITLE
completed frequencyRDD with data frames

### PR DESCRIPTION
**Pull request by akmorrow13**

### CODE REVIEW

**Comment by akmorrow13** [comment]
Jenkins, test this please.

**Comment by akmorrow13** [comment]
Jenkins, test this please.

File: mango-cli/src/main/webapp/WEB-INF/layouts/overall.ssp [hunk 1]
```diff
@@ -27,6 +27,13 @@
 </div>
 
 <% render("templates/sidebar.ssp") %>
+<script src="js/utilities.js"></script>
```
Comment (erictu) [inline mango-cli/src/main/webapp/WEB-INF/layouts/overall.ssp]: shouldn't be repeated

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 2]
```diff
@@ -121,6 +121,7 @@ object VizReads extends BDGCommandCompanion with Logging {
 
   //Correctly shuts down the server
   def quit() {
+    // TODO: save frequencies
```
Comment (erictu) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Has this been done yet?

File: mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala [hunk 3]
```diff
@@ -177,29 +159,30 @@ class AlignmentRecordMaterialization(s: SparkContext,
...
         })
-        rememberValues(region, ks)
+
+        // key data by ReferenceRegion
+        val keyedAlignments = alignments
+          .map(r => (ReferenceRegion(r), r))
+          .partitionBy(partitioner)
+
+        // map to CalculatedAlignmentRecord
+        val data = keyedAlignments.map(r => (r._1, CalculatedAlignmentRecord(r._2, MismatchLayout(r._2, ref.get, reg))))
```
Comment (erictu) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala]: two values for data?

File: mango-core/src/main/scala/org/bdgenomics/mango/models/FrequencyMaterialization.scala [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  /*
+   * Adds coverage file to fileMap. If coverage file is found, file is
+   * added as coverage. Otherwise, file is added as raw alignment data.
+   *
+   * @parameter filePath: String
+   * @parameter sampleId: String
+   */
+  def loadSample(filePath: String, sampleId: String): Unit = {
+    val coveragePath = filePath + ".coverage.adam"
+    val coverageFile = new File(coveragePath)
```
Comment (erictu) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/FrequencyMaterialization.scala]: This won't work on remote filesystems. Use isLocal

File: mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala [hunk 5]
```diff
@@ -213,8 +196,41 @@ object AlignmentRecordMaterialization {
...
+    } else {
+      sc.loadIndexedBam(fp, region)
+    }
+  }
+
+  def loadAdam(sc: SparkContext, region: ReferenceRegion, fp: String): RDD[AlignmentRecord] = {
+    val pred: FilterPredicate = ((LongColumn("end") >= region.start) && (LongColumn("start") <= region.end) && (BinaryColumn("contigName") === (region.referenceName)))
+    val proj = Projection(AlignmentRecordField.contigName, AlignmentRecordField.mapq, AlignmentRecordField.readName, AlignmentRecordField.start,
+      AlignmentRecordField.end, AlignmentRecordField.sequence, AlignmentRecordField.cigar, AlignmentRecordField.readNegativeStrand, AlignmentRecordField.readPaired, AlignmentRecordField.recordGroupSample)
+    sc.loadParquetAlignments(fp, predicate = Some(pred), projection = Some(proj)).rdd
```
Comment (erictu) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala]: why the need for .rdd? Doesn't `sc.loadParquetAlignments` already return an RDD

File: mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala (hunk 3 above)
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala]: what does this comment mean?

**Comment by erictu** [comment]
Thanks @akmorrow13

**Pull request merged**